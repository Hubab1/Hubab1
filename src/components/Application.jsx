import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { buildRoute, getInitialPage } from 'utils/routeNavigation';
import { Routes } from 'constants.js';
import ProfileContainer from 'components/profile/ProfileContainer';
import TermsPage from 'components/TermsPage';
import AppContext from 'contexts/AppContext';
export default class Application extends React.Component {
    static contextType = AppContext;
    componentDidMount () {
        const { history, match } = this.props;
        const initialPage = getInitialPage(this.context.communityId, this.props.profile);
        history.replace(initialPage);
        history.replace('terms')
    }
    
    render () {
        const { match } = this.props;
        return (
            <Switch>
                <Route path={buildRoute(match.url, Routes.PROFILE)} component={ProfileContainer} />
                <Route path={buildRoute(match.url, Routes.TOS)} component={TermsPage}/>
            </Switch>
        )
    }
}