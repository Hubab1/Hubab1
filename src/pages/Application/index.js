import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';

import { ROUTES } from 'constants/constants';
import ApplicationCompletePage from './pages/ApplicationComplete';
import ApplicationApprovedPage from './pages/ApplicationApproved';
import ApplicationDeniedPage from './pages/ApplicationDenied';
import ApplicationCancelledPage from './pages/ApplicationCancelled';

class ApplicationContainer extends Component {
    render() {
        return (
            <Switch>
                <Route path={ROUTES.APP_COMPLETE} component={ApplicationCompletePage} />
                <Route path={ROUTES.APP_APPROVED} component={ApplicationApprovedPage} />
                <Route path={ROUTES.APP_DENIED} component={ApplicationDeniedPage} />
                <Route path={ROUTES.APP_CANCELLED} component={ApplicationCancelledPage} />
            </Switch>
        );
    }
}

export default ApplicationContainer;
