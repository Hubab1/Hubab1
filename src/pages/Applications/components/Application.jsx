import React, { useCallback, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { isEmpty } from 'lodash';
import clsx from 'clsx';
import {
    makeStyles,
    Card,
    CardHeader,
    CardContent,
    Collapse,
    IconButton,
    Typography,
    Divider,
} from '@material-ui/core';
import { ExpandMore as ExpandMoreIcone } from '@material-ui/icons';

import {
    APPLICATION_STATUSES_LABELS,
    APPLICANT_ROLE_LABELS,
    APPLICATION_STATUSES_COLORS,
    APPLICATION_STATUS_DENIED,
} from 'constants/constants';
import ActionButton from 'common-components/ActionButton/ActionButton';
import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import { fetchRenterProfile, selectors } from 'reducers/renter-profile';

const useStyles = makeStyles((theme) => ({
    root: {
        marginBottom: theme.spacing(2),
        '&:last-child': {
            marginBottom: 0,
        },
    },
    expand: {
        transform: 'rotate(0deg)',
        marginLeft: 'auto',
        transition: theme.transitions.create('transform', {
            duration: theme.transitions.duration.shortest,
        }),
    },
    expandOpen: {
        transform: 'rotate(180deg)',
    },
    divider: {
        margin: theme.spacing(0, 2),
    },
    typography: {
        color: '#828796',
        marginBottom: theme.spacing(0.5),

        '& span': {
            color: 'black',
        },
    },
    title: {
        color: '#000',
    },
    applicationId: {
        color: '#828796',
        display: 'block',
        marginTop: theme.spacing(2),
    },
}));

export function Application({ application = {}, isActive = true, fetchRenterProfile, initialPage, history }) {
    const { id, status, lease_start_date, lease_term, fees_breakdown, role, unit, community } = application;
    const classes = useStyles();
    const [expanded, setExpanded] = useState(isActive);
    const title = unit?.unit_number ? `${community.display_name}, #${unit.unit_number}` : community.display_name;

    const handleExpandClick = useCallback(() => {
        setExpanded(!expanded);
    }, [expanded]);

    const [appSelected, setAppSelected] = useState(false);

    useEffect(() => {
        if (initialPage && appSelected) {
            setAppSelected(false);
            history.push(initialPage);
        }
    }, [initialPage, appSelected, history]);

    const handleApplicationClick = async (id) => {
        await fetchRenterProfile(id);
        setAppSelected(true);
    };

    if (isEmpty(application)) {
        return null;
    }

    const showOpenApplicationButton = isActive || application.status === APPLICATION_STATUS_DENIED;

    return (
        <Card className={classes.root} elevation={2}>
            <CardHeader
                data-testid="header"
                title={
                    <Typography className={clsx(classes.typography, classes.title)} variant="body1">
                        <b>{title}</b>
                    </Typography>
                }
                subheader={
                    <Typography className={classes.typography} variant="body1">
                        Application Status:{' '}
                        <span style={{ color: APPLICATION_STATUSES_COLORS[status] }}>
                            {APPLICATION_STATUSES_LABELS[status]}
                        </span>
                    </Typography>
                }
                action={
                    <IconButton
                        className={clsx(classes.expand, {
                            [classes.expandOpen]: expanded,
                        })}
                        onClick={handleExpandClick}
                    >
                        <ExpandMoreIcone />
                    </IconButton>
                }
            />
            <Collapse in={expanded}>
                <Divider className={classes.divider} />
                <CardContent data-testid="content">
                    <Typography className={classes.typography} variant="body1">
                        Move in Date:{' '}
                        <span>{lease_start_date ? moment(lease_start_date).format('MM/DD/YYYY') : '---'}</span>
                    </Typography>
                    <Typography className={classes.typography} variant="body1">
                        Lease Term: <span>{lease_term ? lease_term : '---'}</span>
                    </Typography>
                    <Typography className={classes.typography} variant="body1">
                        Monthly Rent:{' '}
                        <span>
                            {fees_breakdown?.monthly_fees?.total ? `$${fees_breakdown.monthly_fees.total}` : '---'}
                        </span>
                    </Typography>
                    <Typography className={classes.typography} variant="body1">
                        Role: <span>{`${APPLICANT_ROLE_LABELS[role]}`}</span>
                    </Typography>
                    <Typography className={classes.applicationId} variant="caption">
                        Application ID <span>{id}</span>
                    </Typography>
                    {showOpenApplicationButton && (
                        <ActionButton
                            variant="outlined"
                            marginTop={20}
                            marginBottom={10}
                            onClick={() => handleApplicationClick(id)}
                        >
                            Go To Application
                        </ActionButton>
                    )}
                </CardContent>
            </Collapse>
        </Card>
    );
}

Application.propTypes = {
    application: PropTypes.shape({
        id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
        status: PropTypes.number.isRequired,
        lease_start_date: PropTypes.string,
        lease_term: PropTypes.number,
        fees_breakdown: PropTypes.object,
        role: PropTypes.number.isRequired,
        unit: PropTypes.object,
        community: PropTypes.object.isRequired,
    }),
    isActive: PropTypes.bool,
    initialPage: PropTypes.string,
    communityId: PropTypes.string,
    fetchRenterProfile: PropTypes.func,
    history: PropTypes.object,
};

const mapStateToProps = (state) => ({
    communityId: state.siteConfig.basename,
    initialPage: selectors.selectDefaultInitialPage(state),
});

const mapDispatchToProps = {
    fetchRenterProfile,
};

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Application));
