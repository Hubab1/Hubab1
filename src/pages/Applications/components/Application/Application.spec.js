import React from 'react';
import { shallow } from 'enzyme';

import {
    ACTIVE_APPLICATION_STATUSES,
    PAST_APPLICATION_STATUSES,
    APPLICANT_ROLE_VALUES,
    APPLICATION_STATUSES_LABELS,
    APPLICANT_ROLE_LABELS,
    APPLICATION_STATUSES_COLORS,
} from 'app/constants';
import Application from './Application';

describe('Application', () => {
    it('renders active application', () => {
        const application = {
            id: 1,
            status: ACTIVE_APPLICATION_STATUSES[0],
            lease_start_date: new Date('01-01-2020'),
            lease_term: 12,
            fees_breakdown: {
                monthly_fees_v2: {
                    total: '1.500',
                },
            },
            unit: {
                unit_number: '12',
            },
            community: {
                display_name: 'Community Alpha',
            },
        };

        const wrapper = shallow(
            <Application application={application} role={APPLICANT_ROLE_VALUES.ROLE_PRIMARY_APPLICANT} isActive />
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders past application', () => {
        const application = {
            id: 1,
            status: PAST_APPLICATION_STATUSES[0],
            lease_start_date: new Date('01-01-2020'),
            lease_term: 12,
            fees_breakdown: {
                monthly_fees_v2: {
                    total: '1.500',
                },
            },
            unit: {
                unit_number: '12',
            },
            community: {
                display_name: 'Community Alpha',
            },
        };

        const wrapper = shallow(
            <Application application={application} role={APPLICANT_ROLE_VALUES.ROLE_CO_APPLICANT} isActive={false} />
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
