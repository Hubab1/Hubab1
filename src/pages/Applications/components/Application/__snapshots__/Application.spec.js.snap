// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application renders active application 1`] = `
<ForwardRef(WithStyles)
  className="makeStyles-root-1"
  elevation={2}
>
  <ForwardRef(WithStyles)
    action={
      <ForwardRef(WithStyles)
        className="makeStyles-expand-2 makeStyles-expandOpen-3"
        onClick={[Function]}
        style={
          Object {
            "display": "inline-flex",
          }
        }
      >
        <UNDEFINED />
      </ForwardRef(WithStyles)>
    }
    subheader={
      <ForwardRef(WithStyles)
        className="makeStyles-typography-5"
        variant="body1"
      >
        Application Status:
         
        <span
          style={
            Object {
              "color": "#000",
            }
          }
        >
          In Progress
        </span>
      </ForwardRef(WithStyles)>
    }
    title={
      <ForwardRef(WithStyles)
        className="makeStyles-typography-5 makeStyles-title-6"
        variant="body1"
      >
        <b>
          Community Alpha, #12
        </b>
      </ForwardRef(WithStyles)>
    }
  />
  <ForwardRef(WithStyles)
    in={true}
  >
    <ForwardRef(WithStyles)
      className="makeStyles-divider-4"
    />
    <ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        className="makeStyles-typography-5"
        variant="body1"
      >
        Move in Date: 
        <span>
          01/01/2020
        </span>
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        className="makeStyles-typography-5"
        variant="body1"
      >
        Lease Term: 
        <span>
          12
        </span>
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        className="makeStyles-typography-5"
        variant="body1"
      >
        Montly Rent: 
        <span>
          $
          1.500
        </span>
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        className="makeStyles-typography-5"
        variant="body1"
      >
        Role: 
        <span>
          Main Applicant
        </span>
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        className="makeStyles-typography-5 makeStyles-applicationId-7"
        variant="caption"
      >
        Application ID: 
        <span>
          1
        </span>
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>
</ForwardRef(WithStyles)>
`;

exports[`Application renders past application 1`] = `
<ForwardRef(WithStyles)
  className="makeStyles-root-1"
  elevation={2}
>
  <ForwardRef(WithStyles)
    action={
      <ForwardRef(WithStyles)
        className="makeStyles-expand-2"
        onClick={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
      >
        <UNDEFINED />
      </ForwardRef(WithStyles)>
    }
    subheader={
      <ForwardRef(WithStyles)
        className="makeStyles-typography-5"
        variant="body1"
      >
        Application Status:
         
        <span
          style={
            Object {
              "color": "#000",
            }
          }
        >
          Completed
        </span>
      </ForwardRef(WithStyles)>
    }
    title={
      <ForwardRef(WithStyles)
        className="makeStyles-typography-5 makeStyles-title-6"
        variant="body1"
      >
        <b>
          Community Alpha, #12
        </b>
      </ForwardRef(WithStyles)>
    }
  />
</ForwardRef(WithStyles)>
`;
